### PR TITLE
Fixed reading the callchain record and added a flag

### PR DESCRIPTION
**Pull request by florian-kuebler**
Fixed a bug, when reading a callchain record, which resulted in a segfault.
Further added a flag (frame_pointer_unwinding) to enable frame-pointer based unwinding as long as there is no UI option for this.

### CODE REVIEW

File: OrbitCore/LinuxTracingHandler.cpp [hunk 1]
```diff
@@ -16,6 +16,10 @@
 ABSL_FLAG(uint16_t, sampling_rate, 1000,
           "Frequency of callstack sampling in samples per second");
 
+// TODO: Remove this flag once we have an ui option to specify.
```
Comment (dimitry-) [inline OrbitCore/LinuxTracingHandler.cpp:19]: please move this flag to main.cpp and then propagate it to this class - when this flag is set from UI it should not modify global but call a corresponding setter in this object.

Comment (florian-kuebler) [reply OrbitCore/LinuxTracingHandler.cpp:19]: Well, how should this look like? Should I move the flag to OrbitGl/main.cpp? And then add a setter method for the unwinding method?

Comment (florian-kuebler) [reply OrbitCore/LinuxTracingHandler.cpp:19]: Having the setter, would be then probably be multiple setter, right? Access to this instance of LinuxTracingHandler has currently OrbitAsioServer. So we would need to propagate that flag from Client to Service first.

Comment (dimitry-) [reply OrbitCore/LinuxTracingHandler.cpp:19]: OrbitService main.cpp for now...

File: OrbitLinuxTracing/PerfEventReaders.cpp [hunk 2]
```diff
@@ -79,6 +79,9 @@ std::unique_ptr<CallchainSamplePerfEvent> ConsumeCallchainSamplePerfEvent(
       &event->ring_buffer_record.sample_id,
       offsetof(perf_event_callchain_sample, sample_id));
 
+  ring_buffer->ReadValueAtOffset(&event->ring_buffer_record.nr,
```
Comment (dpallotti) [inline OrbitLinuxTracing/PerfEventReaders.cpp:82]: It is confusing that you read this twice. The constructor of `CallchainSamplePerfEvent` should probably set it. Or just use `event->ring_buffer_record.nr = nr` (but I prefer the first).

File: OrbitCore/LinuxTracingHandler.cpp [hunk 3]
```diff
@@ -16,6 +16,10 @@
...
+ABSL_FLAG(bool, frame_pointer_unwinding, false,
+          "Use frame pointer information for unwinding");
```
Comment (dpallotti) [inline OrbitCore/LinuxTracingHandler.cpp:21]:
```suggestion
          "Use frame pointers for unwinding");
```

File: OrbitCore/LinuxTracingHandler.cpp (hunk 1 above)
Comment (florian-kuebler) [reply OrbitCore/LinuxTracingHandler.cpp]: Okay, I moved the flag accordingly.

File: OrbitLinuxTracing/PerfEventReaders.cpp (hunk 2 above)
Comment (florian-kuebler) [reply OrbitLinuxTracing/PerfEventReaders.cpp]: I changed it the way, that the constructor sets the field.

File: OrbitCore/LinuxTracingHandler.cpp (hunk 1 above)
Comment (dpallotti) [reply OrbitCore/LinuxTracingHandler.cpp]: I don't know @dimitry- , now we have a `sampling_method` parameter everywhere and I can't say that I like it.

Comment (dimitry-) [reply OrbitCore/LinuxTracingHandler.cpp]: it is in c-tor which is fine. And the services need a way to propagate it. I think this is fine, at least there is an explicit flag in LinuxTracingHandler which could now be set from UI.

Comment (dpallotti) [reply OrbitCore/LinuxTracingHandler.cpp]: Then should we just propagate the whole `TracingOptions` struct?

Comment (dimitry-) [reply OrbitCore/LinuxTracingHandler.cpp]: makes sense.

Comment (florian-kuebler) [reply OrbitCore/LinuxTracingHandler.cpp]: Changed it, such that the whole `TracingOptions` will be passed.

File: OrbitService/main.cpp [hunk 4]
```diff
@@ -12,6 +14,10 @@ ABSL_FLAG(uint16_t, asio_port, 44766, "Asio TCP server port");
 ABSL_FLAG(std::string, grpc_server_address, "0.0.0.0:44755",
           "Grpc server address");
 
+// TODO: Remove this flag once we have an ui option to specify.
```
Comment (dpallotti) [inline OrbitService/main.cpp:17]:
```suggestion
// TODO: Remove this flag once we have a ui option to specify.
```

File: OrbitService/main.cpp [hunk 5]
```diff
@@ -40,7 +46,17 @@ int main(int argc, char** argv) {
   std::string grpc_server_address = absl::GetFlag(FLAGS_grpc_server_address);
   uint16_t asio_port = absl::GetFlag(FLAGS_asio_port);
 
+  LinuxTracing::TracingOptions tracing_options;
+  tracing_options.trace_context_switches = GParams.m_TrackContextSwitches;
+
+  bool use_frame_pointer_unwinding =
+      absl::GetFlag(FLAGS_frame_pointer_unwinding);
+  if (use_frame_pointer_unwinding) {
```
Comment (dpallotti) [inline OrbitService/main.cpp:54]: Use `absl::GetFlag(FLAGS_frame_pointer_unwinding)` directly here?

File: OrbitService/main.cpp [hunk 6]
```diff
@@ -40,7 +46,17 @@ int main(int argc, char** argv) {
   std::string grpc_server_address = absl::GetFlag(FLAGS_grpc_server_address);
   uint16_t asio_port = absl::GetFlag(FLAGS_asio_port);
 
+  LinuxTracing::TracingOptions tracing_options;
+  tracing_options.trace_context_switches = GParams.m_TrackContextSwitches;
```
Comment (dpallotti) [inline OrbitService/main.cpp:50]: This would only work on the client side, you can finally get rid of this.

File: OrbitService/main.cpp [hunk 7]
```diff
@@ -1,7 +1,9 @@
 #include <csignal>
 #include <iostream>
 
+#include "OrbitLinuxTracing/TracingOptions.h"
 #include "OrbitService.h"
+#include "Params.h"
```
Comment (dpallotti) [inline OrbitService/main.cpp:6]: Remove when removing `tracing_options.trace_context_switches = GParams.m_TrackContextSwitches;`.

Comment (florian-kuebler) [reply OrbitService/main.cpp:6]: Done

File: OrbitService/main.cpp (hunk 4 above)
Comment (florian-kuebler) [reply OrbitService/main.cpp:17]: Applied with last commit. Thanks.

File: OrbitService/main.cpp (hunk 5 above)
Comment (florian-kuebler) [reply OrbitService/main.cpp:54]: Sure.

File: OrbitService/main.cpp (hunk 6 above)
Comment (florian-kuebler) [reply OrbitService/main.cpp:50]: Sure.

**Pull request merged**